src/efftester.ml

```diff
@@ -740,6 +827,12 @@ module GeneratorsWithContext (Ctx : Context) = struct
     | Float -> Gen.map (fun f -> LitFloat f) float_gen
     | Bool -> Gen.map (fun b -> LitBool b) Gen.bool
     | String -> Gen.map (fun s -> LitStr s) string_gen
+    | Option t' ->
+      let open Gen in
+      frequencyl [ (2, `Some); (1, `None) ] >>= fun opt ->
+      (match opt with
+      | `Some -> literal_gen t' eff (size / 2) >|= fun e -> LitOption (Some e)
```

Comment:
Note: here there is only one recursive call to `literal_gen`, so you don't need to divide the size by 2, I think `size - 1` would be fine -- and would generate larger terms.

Note also: in the `List` case, there is protection for the case where `size = 0`, to not try to generate a non-empty list. I think you could have done the same here and generated only `None` in that case. Anyway I think this whole code block will disappear if you remove `LitOption`.